modules/objdetect/include/opencv2/objdetect/checker_detector.hpp

```diff
@@ -0,0 +1,222 @@
+ *                                  (default 153).
+ * - int adaptiveThreshWinSizeStep : increments from adaptiveThreshWinSizeMin to
+ *                                   adaptiveThreshWinSizeMax during the
+ *                                   thresholding (default 16).
+ * - double adaptiveThreshConstant : constant for adaptive thresholding before
+ *                                   finding contours (default 7)
+ * - double minContoursAreaRate : determine minimum area for marker contour to
+ *                                be detected. This is defined as a rate respect
+ *                                to the area of the input image. Used only if
+ *                                neural network is used (default 0.003).
+ * - double minContoursArea : determine minimum area for marker contour to be
+ *                            detected. This is defined as the actual area. Used
+ *                            only if neural network is not used (default 100).
+ * - double confidenceThreshold : minimum confidence for a bounding box detected
+ *                                by neural network to classify as
+ *                                detection.(default 0.5)
+ *                                (0<=confidenceThreshold<=1)
+ * - double minContourSolidity : minimum solidity of a contour for it be
+ *                               detected as a square in the chart. (default
+ *                               0.9).
+ * - double findCandidatesApproxPolyDPEpsMultiplier : multipler to be used in
+ *                                                    cv::ApproxPolyDP function
+ *                                                    (default 0.05)
+ * - int borderWidth : width of the padding used to pass the inital neural
+ *                     network detection in the succeeding system.(default 0)
+ * - float B0factor : distance between two neighbours squares of the same chart.
+ *                    Defined as the ratio between distance and large dimension
+ *                    of square (default 1.25)
+ * - float maxError : maximum allowed error in the detection of a chart.
+ *                    default(0.1)
+ * - int minContourPointsAllowed : minium points in a detected contour.
+ *                                 default(4)
+ * - int minContourLengthAllowed : minimum length of a countour. default(100)
+ * - int minInterContourDistance : minimum distance between two contours.
+ *                                 default(100)
+ * - int minInterCheckerDistance : minimum distance between two checkers.
+ *                                 default(10000)
+ * - int minImageSize : minimum size of the smaller dimension of the image.
+ *                      default(1000)
+ * - unsigned minGroupSize : minimum number of a squared of a chart that must be
+ *                           detected. default(4)
+ */
+struct CV_EXPORTS_W DetectorParametersMCC
+{
+
+    DetectorParametersMCC();
+
+    CV_WRAP static Ptr<DetectorParametersMCC> create();
```

Comment:
We do not use Ptr<> for parameter structures. Just const & in detector::create()